samza-test/src/test/java/org/apache/samza/test/table/TestLocalTable.java

```diff
@@ -146,10 +147,90 @@ public void testStreamTableJoin() throws Exception {
     runner.run(app);
     runner.waitForFinish();
 
+    assertEquals(count * partitionCount, received.size());
     assertEquals(count * partitionCount, joined.size());
     assertTrue(joined.get(0) instanceof EnrichedPageView);
   }
 
+  @Test
+  public void testDualStreamTableJoin() throws Exception {
+
+    List<Profile> sentToProfileTable = new LinkedList<>();
+    List<PageView> sentToPageViewTable = new LinkedList<>();
+    List<EnrichedPageView> joinedProfiles = new LinkedList<>();
+    List<EnrichedPageView> joinedPageViews = new LinkedList<>();
+
+    KVSerde<Integer, Profile> profileKVSerde = KVSerde.of(new IntegerSerde(), new ProfileJsonSerde());
+    KVSerde<Integer, PageView> pageViewKVSerde = KVSerde.of(new IntegerSerde(), new PageViewJsonSerde());
+
+    MyJoinFunction1 joinFn1 = new MyJoinFunction1();
+    MyJoinFunction2 joinFn2 = new MyJoinFunction2();
+
+    int count = 10;
+    PageView[] pageViews = TestTableData.generatePageViews(count);
+    Profile[] profiles = TestTableData.generateProfiles(count);
+
+    int partitionCount = 4;
+    Map<String, String> configs = getBaseJobConfig();
+
+    configs.put("streams.PageView.samza.system", "test");
+    configs.put("streams.PageView.source", Base64Serializer.serialize(pageViews));
+    configs.put("streams.PageView.partitionCount", String.valueOf(partitionCount));
+
+    configs.put("streams.Profile.samza.system", "test");
+    configs.put("streams.Profile.source", Base64Serializer.serialize(profiles));
+    configs.put("streams.Profile.partitionCount", String.valueOf(partitionCount));
+
+    final LocalApplicationRunner runner = new LocalApplicationRunner(new MapConfig(configs));
+    final StreamApplication app = (streamGraph, cfg) -> {
+
+      Table<KV<Integer, Profile>> profileTable = streamGraph.getTable(new InMemoryTableDescriptor("t1")
+          .withSerde(KVSerde.of(new IntegerSerde(), new ProfileJsonSerde())));
+
+      Table<KV<Integer, PageView>> pageViewTable = streamGraph.getTable(new InMemoryTableDescriptor("t2")
+          .withSerde(KVSerde.of(new IntegerSerde(), new PageViewJsonSerde())));
+
+      MessageStream<Profile> profileStream = streamGraph.getInputStream("Profile", new NoOpSerde<Profile>());
+      MessageStream<PageView> pageViewStream = streamGraph.getInputStream("PageView", new NoOpSerde<PageView>());
+
+      profileStream
+          .map(m -> {
+              sentToProfileTable.add(m);
+              return new KV(m.getMemberId(), m);
+            })
+          .sendTo(profileTable);
```

Comment:
My original comment is actually testing the case with: 
  profileStream....sendTo(profileTable);
  pageViewStream....sendTo(profileTable);
  inStream....join(profileTable);
  inStream2...join(profileTable);

Reply:
I don't think pageView...sendTo(profileTable) would work, let's clarify the test case